mission/functions/core/ui/taskroster/fn_tr_mainInfo_show.sqf

```diff
@@ -12,101 +12,323 @@
-	case "GreenHornets":{"STR_vn_mf_taskRoster_Main_GreenHornetsTeamDescription"};
-	case "ACAV":{"STR_vn_mf_taskRoster_Main_ArmouredCavalryDescription"};
-	default {""};
+private _groupDescription = getText(_groupConfig >> "description");
+private _groupMembers = missionNamespace getVariable [_groupID, []];
+private _traitConfigs = "true" configClasses (missionConfigFile >> "gamemode" >> "traits");
+
+/////////////////////////////////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////////////////////////
+// DATA TRANSFORMATION FOR DISPLAY
+
+// list of traits with player counts
+private _traitsTeam = _traitConfigs select {(configName _x) isNotEqualTo "vn_artillery"} apply {
+
+	private _traitConfig = _x;
+	private _traitName = configName _x;
+	private _traitIcon = getText (_traitConfig >> "icon");
+	private _traitText = (getText (_traitConfig >> "text") call para_c_fnc_localize);
+	private _traitPlayerCount = count (
+		_groupMembers select {_x getUnitTrait _traitName}
+	);
+	private _traitPlayerCountLimit = getNumber (_groupConfig >> "rolelimits" >> _traitName);
+
+	private _hmap = createHashMapFromArray [
+		["name", _traitName],
+		["icon", _traitIcon],
+		["text", _traitText],
+		["playerCount", _traitPlayerCount],
+		["playerLimit", _traitPlayerCountLimit]
+	];
+
+	_hmap
 };
 
-VN_TR_MAININFO_TXT_MID_CTRL ctrlSetStructuredText parseText format [localize _text, format ["<img size='5' color='#FFFFFF' image='%1'/>", _groupIcon]];
-VN_TR_MAININFO_TXT_MID_CTRL ctrlSetPositionH (ctrlTextHeight VN_TR_MAININFO_TXT_MID_CTRL);
-VN_TR_MAININFO_TXT_MID_CTRL ctrlCommit 0;
+// players list with roles
+private _teamTraits = _groupMembers apply {
+	private _player = _x;
 
-private _traitConfigs = "true" configClasses (missionConfigFile >> "gamemode" >> "traits");
+	private _traitsIconsTooltips = _traitsTeam select {
+		_player getUnitTrait (_x get "name")
+	} apply {
+		[_x get "icon", _x get "text"]
+	};
+
+	private _hmap = createHashMapFromArray [
+		["playerName", name _player],
+		["traits", _traitsIconsTooltips]
+	];
+	_hmap
+};
+
+private _traitsPlayer = _traitConfigs select {
+	(player getUnitTrait (configName _x))
+} apply {
+
+	private _traitConfig = _x;
+	private _traitName = configName _x;
+	private _traitIcon = getText (_traitConfig >> "icon");
+	private _traitText = (getText (_traitConfig >> "text") call para_c_fnc_localize);
+
+	private _traitTasks = switch _traitName do {
+		case "medic":{localize "STR_vn_mf_taskRoster_Main_medicTasks"};
+		case "engineer":{localize "STR_vn_mf_taskRoster_Main_engineerTasks"};
+		case "explosiveSpecialist":{localize "STR_vn_mf_taskRoster_Main_explosive_specialistTasks"};
+		case "radioOperator": { localize "STR_vn_mf_radioOperatorTasks" };
+		default {""};
+	};
+
+	private _hmap = createHashMapFromArray [
+		["name", _traitName],
+		["icon", _traitIcon],
+		["text", _traitText],
+		["desc", _traitTasks]
+	];
+
+	_hmap
```

Comment:
player's traits -- is this needed in here anymore, seeing as the player info (roles) has been split from teams info?